### PR TITLE
Mobile API refactor.

### PR DESCRIPTION
**Pull request by cpasillas**

### CODE REVIEW

File: mobile/proxy.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+	vh.SetResponseVerifier(mod)
+
+	m.handle("/verify", vh)
+
+	// Reset verifications.
+	rh := verify.NewResetHandler()
+	rh.SetRequestVerifier(mod)
+	rh.SetResponseVerifier(mod)
+	m.handle("/verify/reset", rh)
+
```
Comment (bramhaghosh) [inline mobile/proxy.go]: can we add a handler to expose the marbl logs?

File: mobile/proxy.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	mlog.Infof("mobileproxy: proxy shut down")
+}
+
+// SetLogLevel sets the Martian log level (Silent = 0, Error, Info, Debug), controlling which Martian
+// log calls are displayed in the console
+func SetLogLevel(l int) {
+	mlog.SetLevel(l)
+}
+
+func init() {
```
Comment (bramhaghosh) [inline mobile/proxy.go]: move the init() to the top of the file

File: mobile/proxy.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	m.proxy = martian.NewProxy()
+	m.mux = http.NewServeMux()
+
+	if m.Cert != "" && m.Key != "" {
+		tlsc, err := tls.X509KeyPair([]byte(m.Cert), []byte(m.Key))
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		mlog.Debugf("mobileproxy: loaded cert and key")
```
Comment (bramhaghosh) [inline mobile/proxy.go]: log lines should start
mobile: 
not 
mobileproxy:

Comment (cpasillas) [inline mobile/proxy.go]: Done

File: mobile/proxy.go (hunk 2 above)
Comment (cpasillas) [inline mobile/proxy.go]: Done

File: mobile/proxy.go (hunk 1 above)
Comment (cpasillas) [inline mobile/proxy.go]: Done

**Pull request merged**